src/ocaml/typing/typecore.ml

```diff
@@ -2463,7 +2464,7 @@ let check_partial_application statement exp =
             | Texp_setinstvar _ | Texp_override _ | Texp_assert _
             | Texp_lazy _ | Texp_object _ | Texp_pack _ | Texp_unreachable
             | Texp_extension_constructor _ | Texp_ifthenelse (_, _, None)
-            | Texp_function _ ->
+            | Texp_function _ | Texp_hole ->
```

Comment:
I'm not sure that's right, I think this could result in spurious warnings (spurious because the code hasn't been written yet).
I think `| Texp_hole -> ()` would probably be better.